### PR TITLE
Fix inappropriate type (Not a directory) error in Linux for File path…

### PR DESCRIPTION
**Pull request by rnjtranjan**
…s in recursive mode

Fix for #1307

### CODE REVIEW

File: src/Streamly/Internal/FileSystem/Event/Windows.hs [hunk 1]
```diff
@@ -12,7 +12,7 @@
 --
 -- =Overview
 --
--- Use 'watchTrees'or 'watchPaths' with a list of file system dir paths you 
+-- Use 'watchTrees'or 'watchPaths' with a list of file system dir paths you
```
Comment (harendra-kumar) [inline src/Streamly/Internal/FileSystem/Event/Windows.hs:15]: We no longer have these APIs. We have `watch` and `watchRecursive`.

**Review by harendra-kumar — CHANGES_REQUESTED** [review]
Can you also enable the commented test for the case when root path is a file?

**Comment by rnjtranjan** [comment]
> 

For windows test for the case when root path is a file is not valid as windows API doesn't support it.
For Linux nested file create event times out due to the event is lost.

**Review by rnjtranjan — COMMENTED** [review]
Looks good

**Pull request merged**